worker/worker.go

```diff
@@ -168,6 +173,39 @@ func (w *Worker) EmitEvent(event ipc.WorkerEventName, messageID string, message
 	return w.conn.Emit(dbus.ObjectPath(path.Join("/com/redhat/Yggdrasil1/Worker1", w.directive)), "com.redhat.Yggdrasil1.Worker1.Event", args...)
 }
 
+// cancel implements com.redhat.Yggdrasil1.Worker1.Cancel method by calling the
+// worker's cancelRXFunc in a goroutine.
+func (w *Worker) cancel(addr string, id string, cancelID string) *dbus.Error {
+	// If worker doesn't implements the cancelation function it does nothing
+	if w.cancelRX == nil {
+		log.Debug("worker does not support cancellation messages")
+		return nil
```

Comment:
I wonder if it's worth responding with a D-Bus error here. This would give yggd
some knowledge about whether the cancellation will work; which it in turn could
pass back across the network transport as an event so that message originators
can be made aware of whether an attempted cancellation will or won't work.

Reply:
That makes sense. I think is a good idea that workers reports errors to yggdrasil so it can be aware or some things.